### ISSUE TITLE
RDS deployments

### ISSUE THREAD

**Issue by nidelius** [MicrosoftDocs/OfficeDocs-SkypeForBusiness#512]
Is it possible to add any information on or extra details för using this MSI on RDS Session Host deployments.
Redeploying the MSI seems very unconvenient for a 200user deployment. I understand the intention seems to be a shared desktop or similar. Would it also be possible just do download the teams app the regular way if a user unintentionally uninstalled it from user profile?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7b13006b-d1de-8bc7-e0c6-b7022af0e731
* Version Independent ID: d29dae65-bb28-8767-cfd7-07d2ed5fd460
* Content: [Install Microsoft Teams using MSI](https://docs.microsoft.com/en-us/microsoftteams/msi-deployment)
* Content Source: [Teams/msi-deployment.md](https://github.com/MicrosoftDocs/OfficeDocs-SkypeForBusiness/blob/live/Teams/msi-deployment.md)
* Service: **msteams**
* GitHub Login: @ninadara
* Microsoft Alias: **ninadara**

**Comment by Dominik-D** [comment]
The Teams feature (https://microsoftteams.uservoice.com/forums/555103-public/suggestions/17380159-improve-install-options-install-for-all-users-an?tracking_code=7dc01d1693b0b2d4b2c8a49d6c68d8e4) vote pointed to this document suggesting that there is now an all users install option that does not install into AppData. The current state of the MSI installer is not solving that issue as it still installs Teams into the AppData folder of each user.
This is a highly requested feature. Please re-open the feature request and if possible provide information about ETA. Thanks!

**Comment by kenwith** [comment]
@nidelius and @Dominik-D - There is also a lot of feedback on the same in this issue: https://github.com/MicrosoftDocs/OfficeDocs-SkypeForBusiness/issues/101

I surfacing this to the product team regarding the closing of the User Voice mentioned above. I will keep both Issues posted on updates.

**Comment by Dominik-D** [comment]
The Teams feature vote is still closed. Could you kindly push to reopen?

**Comment by WarrenMicrosoftTeams** [comment]
Thank you for the feedback and our Microsoft Teams feature team have seen your feedback. Allow me to provide a response from the feature team. 

1.	**Why Teams App is NOT installed in Program Files (aka per Machine)?**
The goal was to ensure we provide our users with newest features and fixes as quickly as possible and ensure all users are getting the best experience possible from Microsoft Teams. 

Therefore, we don’t limit the ability to update the software by requiring only IT Admin credentials. Unfortunately, if the product were installed under Program Files, that is exactly what would happen. Software under Program Files requires IT admin credentials and the updates would need to be managed per user which in turn would be a lot of overhead due to the frequency of updates to Microsoft Teams.
 
2.	**Why Not follow Office Installation Practice?**
All Office apps require admin credentials to install & update. Office products are updated at a much slower rate than Teams. We are actively creating updates and new features as quickly as possible and having an admin prompt pop due to updates either several times a week or sometimes daily when in Teams would be frustrating for users. 

Also, most managed deployments of Office happen via SCCM (msi based) which is handled by IT admins. The process is a system service and an install would not prompt the user for admin credentials.  Once the installation is complete all future updates are handled by SCCM. If IT admins were required to handle weekly SCCM for various updates, this too would be frustrating/annoying. 

The overall goal is to try and make the installation as painless as possible while enabling our ability to keep Teams updated in a timely fashion without placing undo burden on IT admins within organizations.

**Comment by CIGAdmin** [comment]
The problem with your resolution is that it runs smack against Software Restriction Policies. Most, if not all, Enterprise setting have prohibited running software out of AppData and Temp folders. Worse, having 100s copies of Teams App on a single RDS server is ridiculous. Thus, Enterprise users will not be allowed to use Teams App until this is corrected. Microsoft needs to think like an Enterprise firm.

**Comment by Dominik-D** [comment]
Thanks for providing feedback Warren. I understand the developers point to roll out updates as quickly as possible. That makes sense for users with non-shared fat clients which may fit quite huge portion of your user base.

However I encourage you to look at this from en enterprise perspective where many scenarios require close restrictions an AppData. In an RDS/VDI environment we cannot allow Teams to be installed on every profile. That will dramatically increase storage cost and user logon times. As an example: Our average profile size is roughly 80 MB. With the Teams app it can increase beyond 500 MB. Given that an enterprise customer will be very willing to live with slower update cycles.

**Comment by kenwith** [comment]
Merging discussion to this issue: https://github.com/MicrosoftDocs/OfficeDocs-SkypeForBusiness/issues/101

**Comment by michalzobec** [comment]
Hello Warren, I do not agree with your arguments.

> 1. **Why Teams App is NOT installed in Program Files (aka per Machine)?**
>    The goal was to ensure we provide our users with newest features and fixes as quickly as possible and ensure all users are getting the best experience possible from Microsoft Teams.
> 
> Therefore, we don’t limit the ability to update the software by requiring only IT Admin credentials. Unfortunately, if the product were installed under Program Files, that is exactly what would happen. Software under Program Files requires IT admin credentials and the updates would need to be managed per user which in turn would be a lot of overhead due to the frequency of updates to Microsoft Teams.

Please inspire with Google Chrome/Chromium. Ask for Chromium/Chrome, or your Microsoft Edge team how make frequent updates of application in Enterprise environment without Administrator rights.

> 1. **Why Not follow Office Installation Practice?**
>    All Office apps require admin credentials to install & update. Office products are updated at a much slower rate than Teams. We are actively creating updates and new features as quickly as possible and having an admin prompt pop due to updates either several times a week or sometimes daily when in Teams would be frustrating for users.
> 
> Also, most managed deployments of Office happen via SCCM (msi based) which is handled by IT admins. The process is a system service and an install would not prompt the user for admin credentials. Once the installation is complete all future updates are handled by SCCM. If IT admins were required to handle weekly SCCM for various updates, this too would be frustrating/annoying.
> 
> The overall goal is to try and make the installation as painless as possible while enabling our ability to keep Teams updated in a timely fashion without placing undo burden on IT admins within organizations.

1. Office use MSI packages to Office 2013. From Office 2016 is used Click-to-Run install engine. Since Office 2016 use our self-update mechanism indenpedent of Microsoft Update. Is possible to disable it. So Office running in Program Files folder, and can update self.
2. Again, please inspire with Google Chrome/Chromium.
3. Weekly updates is maybe frustrating/annoying for SCCM admins, but they are responsible, not you. How many updates we make and how often it's none of your business. therefore, it is not possible for you to decide for them. Therefore, it is not possible for you to decide for them. _Then all those tools and features like WSUS and deferred installation of updates would be not make sense._
4. Make multiple copies of application files is crazy idea not only from point of view from RDS servers admins, but for example shared workstations in education environments, or for example call centers, or other business with shift operations. I talking about scenario where is multiple users (in shifts) on single device.
5. In **real** secure environment with AppLocker, is not possible had any executable files in user profiles. This is suspicious and exploitable. For store of application files is used Program Files folders. Please use only Program Files folders. *Realize that in a security-conscious society, it is absolutely unthinkable for a user to install an application.*
6. It is strange that you, as a Microsoft part, do not behave consistently as others Microsoft's teams. I've been following this behavior for a few years with teams developing cross-platform tools, such as Visual Studio Code. You act like you've never been a part of Microsoft. And because you're part of Microsoft, I find your team's behavior embarrassing. Please contact your own enterprise architecture and security engineers and ask them what you are doing wrong. Then you won't have so many upset people here and everyone will be happy.
7. Your product are great. however, also think not only of application users but also of administrators and listen to their needs. It's great to have an amazing product, but what's the point of deploying it when it's deployed in an environment of hundreds of users.
